File: src/backend/internal/model/entity/author.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

// Author 作者实体
type Author struct {
	ID         uint           `gorm:"primaryKey" json:"id"`
	PaperID    uint           `gorm:"index;not null" json:"paper_id"`
	Name       string         `gorm:"type:varchar(100);not null" json:"name"`
	AuthorType string         `gorm:"type:varchar(20);not null" json:"author_type"`
	Rank       int            `gorm:"not null" json:"rank"`
	Department string         `gorm:"type:varchar(100)" json:"department"`
	UserID     *uint          `gorm:"index" json:"user_id"`
	CreatedAt  time.Time      `json:"created_at"`
	UpdatedAt  time.Time      `json:"updated_at"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`

	// 关联关系
	Paper *Paper `gorm:"foreignKey:PaperID" json:"paper,omitempty"`
	User  *User  `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

// TableName 指定表名
func (Author) TableName() string {
	return "authors"
}

File: src/frontend/src/types/author.ts
// 作者相关类型定义

/**
 * 作者类型枚举
 */
export type AuthorType = 'first_author' | 'co_first_author' | 'corresponding_author' | 'author';

/**
 * 作者接口
 */
export interface Author {
  id: number;
  name: string;
  author_type: AuthorType;
  rank: number;
  department: string;
  user_id?: number;
}

/**
 * 作者表单接口
 */
export interface AuthorForm {
  name: string;
  author_type: AuthorType;
  rank: number;
  department: string;
  user_id?: number;
}

/**
 * 作者类型选项
 */
export const AUTHOR_TYPE_OPTIONS = [
  { label: '第一作者', value: 'first_author' as AuthorType },
  { label: '共同第一作者', value: 'co_first_author' as AuthorType },
  { label: '通讯作者', value: 'corresponding_author' as AuthorType },
  { label: '普通作者', value: 'author' as AuthorType },
];

/**
 * 检查作者类型是否互斥
 */
export const AUTHOR_TYPE_CONFLICTS: Record<AuthorType, AuthorType[]> = {
  first_author: ['co_first_author', 'corresponding_author'],
  co_first_author: ['first_author'],
  corresponding_author: ['first_author'],
  author: [],
};
